logic/overworld.py

```diff
@@ -226,9 +223,9 @@ def __init__(self, options, world_setup, r):
         richard_maze = Location()
         self._addEntrance("richard_house", ukuku_prairie, richard_house, None)
         self._addEntrance("richard_maze", richard_maze, richard_cave, None)
-        if options.owlstatues == "both" or options.owlstatues == "overworld":
-            Location().add(OwlStatue(0x0C6)).connect(richard_maze, r.bush)
-        Location().add(SlimeKey()).connect(richard_maze, AND(OR(SWORD, AND(MAGIC_POWDER, MAX_POWDER_UPGRADE), MAGIC_ROD, POWER_BRACELET, BOOMERANG), SHOVEL))
+        slime_key_area = Location().connect(richard_maze, OR(SWORD, AND(MAGIC_POWDER, MAX_POWDER_UPGRADE), MAGIC_ROD, POWER_BRACELET, BOOMERANG))
+        Location().add(OwlStatue(0x0C6)).connect(slime_key_area, options.owlstatues in ('both', 'overworld'))
```

Comment:
Any particular reason this was changed from an `if` into a `true/false` connection? It doesn't match the other owl statues this way, and might confuse the itempool.

Reply:
Nope, it was just on a whim for aesthetics. But you're right, even then it doesn't fit the rest of the owls.

Just so I understand better, is the item pool concern that there will be a mismatch between item count and places to put the items? I did briefly try generation without owls with that in mind, but didn't run into a problem. I assumed it was safely disconnected from the graph when the condition was false.